proof/refine/ARM/Reply_R.thy

```diff
@@ -41,25 +31,30 @@ lemma replyUnlink_st_tcb_at'_sym_ref:
   apply (wpsimp wp: replyUnlink_st_tcb_at'_wp)
   apply (clarsimp simp: obj_at'_def)
   done
+*)
 
-lemma replyRemoveTCB_st_tcb_at'_Inactive:
-  "\<lbrace>\<lambda>s. tcb_at' tptr s \<longrightarrow> st_tcb_at' test tptr s \<and> test Inactive\<rbrace>
+lemma replyRemoveTCB_st_tcb_at'_Inactive':
+  "\<lbrace>\<top>\<rbrace>
    replyRemoveTCB tptr
-   \<lbrace>\<lambda>_. st_tcb_at' test tptr\<rbrace>"
+   \<lbrace>\<lambda>_. st_tcb_at' ((=) Inactive) tptr\<rbrace>"
   unfolding replyRemoveTCB_def
-  apply (wpsimp wp: hoare_drop_imp (* FIXME: dangerous, but necessary to avoid the goal blowing
-                                             up in size *)
-                    hoare_vcg_if_lift hoare_vcg_all_lift replyUnlink_st_tcb_at'_Inactive
-              simp: cleanReply_def
-         split_del: if_split)
+  apply (wpsimp wp: replyUnlink_st_tcb_at')
+  apply (clarsimp simp: pred_tcb_at'_def)
+  done
+
```

Comment:
This is the lemma with `(=) Inactive` and `pred_tcb’_weakenE`. When the `t` in the postcondition is the same as the argument to `replyRemoveTCB`, which was the case in the original lemma, this is what we need. And this is I believe the same thing as what Matt was saying.